include/h2o/socketpool.h

```diff
@@ -40,6 +40,12 @@ typedef enum en_h2o_socketpool_target_type_t {
     H2O_SOCKETPOOL_TYPE_SOCKADDR
 } h2o_socketpool_target_type_t;
 
+/* if there's any other conf for specific balancer, create a subclass */
+typedef struct st_h2o_socketpool_target_conf_t {
+    /* weight for load balancer */
+    size_t weight;
```

Comment:
I would appreciate it if we could use a smaller number space independent to the architecture and have a sensible default. One potential choice could be to use a minimum of `1` and a maximum of `256` with a default value of `16`, which is the same as the weight model of HTTP/2.

By changing the code as such, we could potentially reuse the O(1) scheduler for distributing the load (note: we plan to introduce another scheduler, so the reuse should be proposed in a different PR).

Or does the maximum load ratio of 1:256 sound too small to you?

Reply:
So how about using uint8 with mapping 0-255 to 1-256 or using uint16 but checking if weight is in the [1, 256] domain?

Reply:
@zlm2012
> So how about using uint8 with mapping 0-255 to 1-256

SGTM.

Reply:
And for the default value, would it be kind of confusing to set default weight as 16 where other applications like haproxy & nginx use default weight 1? For example, a user wanted a specific backend to process doubled requests of others then assigned weight 2, but it would turn out to be 1/8 requests of others.

Reply:
That's a good suggestions. Let's go for default 1 and max 256; the same as haproxy.